### ISSUE TITLE
[ntlm.py] using the hash to connect

### ISSUE THREAD

**Issue by AlessandroZ** [CoreSecurity/impacket#206]
Hi, 

I used pywerview (https://github.com/the-useless-one/pywerview) and it prompt an impacket error, so I open the issue here. 

Command line: `python pywerview.py invoke-checklocaladminaccess -w <domain> -u <user> --hashes <hashes> --computername <ip>
`

Here is the stacktrace. 
```
File "/usr/local/lib/python2.7/dist-packages/impacket/smbconnection.py", line 228, in login
    return self._SMBConnection.login(user, password, domain, lmhash, nthash)
  File "/usr/local/lib/python2.7/dist-packages/impacket/smb3.py", line 739, in login
    type3, exportedSessionKey = ntlm.getNTLMSSPType3(auth, respToken['ResponseToken'], user, password, domain, lmhash, nthash)
  File "/usr/local/lib/python2.7/dist-packages/impacket/ntlm.py", line 599, in getNTLMSSPType3
    password = password.decode(encoding)
AttributeError: 'NoneType' object has no attribute 'decode'
```
It supposes that the password variable is always used which it was not the case: 
[https://github.com/CoreSecurity/impacket/blob/master/impacket/ntlm.py#L599](https://github.com/CoreSecurity/impacket/blob/master/impacket/ntlm.py#L599)

Using the password variable, everything is fine: 
Command line: `python pywerview.py invoke-checklocaladminaccess -w <domain> -u <user> --hashes <hashes> --computername <ip> -p <fake_password>`

Thanks again

**Comment by asolino** [comment]
Hey @AlessandroZ 

Whenever you specify hashes, you should set password to `''` instead of `None`.  That's something that can be seen in any of the impacket [examples](https://github.com/CoreSecurity/impacket/tree/master/examples).

Since that is not documented, I added an extra check inside [GetNTLMSSPType3](https://github.com/CoreSecurity/impacket/blob/master/impacket/ntlm.py#L585)(), but the same error could be triggered elsewhere, so please be careful with that.

If it works on your side, please close this issue.

**Comment by AlessandroZ** [comment]
Perfect, problem solved. 

I realized that it could have been solved on the pywerview project (setting default value of password to ''). 

Thanks again for your quick commit.

**Comment by the-useless-one** [comment]
@AlessandroZ, it should now be fixed by the-useless-one/pywerview@7fc01cb9e4e172d053f934925b24e9ecfcd57399.

Cheers,

Y
